tests/NLog.UnitTests/LogMessageFormatterTests.cs

```diff
@@ -34,21 +34,26 @@
 namespace NLog.UnitTests
 {
     using Xunit;
+    using System.Collections.Generic;
 
     public class LogMessageFormatterTests : NLogTestBase
     {
         [Fact]
         public void StructuredLoggingTest()
         {
-            LogEventInfo logEventInfo = new LogEventInfo(LogLevel.Info, string.Empty, "Login request from {Username} for {Application}");
-            logEventInfo.Properties["Username"] = "John";
-            logEventInfo.Properties["Application"] = "BestApplicationEver";
-            logEventInfo.Parameters = new object[] { "Login request from John for BestApplicationEver" };   // Must have parameter to activate MessageFormatter
+            LogEventInfo logEventInfo = new LogEventInfo(LogLevel.Info, string.Empty, string.Empty);
+            MessageTemplate messageTemplate = new MessageTemplate("Login request from {Username} for {Application}", false, new[]
+            {
+                new MessageTemplateParameter("Username", "John", null),
+                new MessageTemplateParameter("Application", "BestApplicationEver", null)
+            });
+            logEventInfo.Parameters = new object[] { "John", "BestApplicationEver", "Login request from John for BestApplicationEver" };
```

Comment:
please split the unit test into 2 tests, or don't use the same values (john, BestApplicationEver).

Reply:
The unit test displays how I'm planning capture the structured parameters from Microsoft Extension Logging.

Reply:
This test is a bit confusing because duplicate values  (john, BestApplicationEver). Is this really needed?  What if we forgot one of the two?